### PR TITLE
[daint-gpu] Split HPX APEX recipe into separate file

### PR DESCRIPTION
**Pull request by msimberg**
While having the APEX build in the same recipe as the main build reduced duplication, actually using the APEX build still required manual intervention (because of the APEX build path not being present in `LD_LIBRARY_PATH`). This splits the two builds up again.

I've also added a variant with networking disabled, which is a common use-case for a few users at CSCS.

### CODE REVIEW

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-APEX.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# contributed by Mikael Simberg, John Biddiscombe and Victor Holanda Rusu (CSCS)
+easyblock = 'CMakeMake'
+
+name = 'HPX'
+version = '1.3.0'
+checksums = ['cd34da674064c4cc4a331402edbd65c5a1f8058fb46003314ca18fa08423c5ad']
```
Comment (lucamar) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-APEX.eb]:
```suggestion
versionsuffix = '-APEX'
checksums = ['cd34da674064c4cc4a331402edbd65c5a1f8058fb46003314ca18fa08423c5ad']
```

**Comment by lucamar** [comment]
Thanks @msimberg: for the standard recipe of `HPX`, please sync with @sebkelle1

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-nonetworking.eb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# contributed by Mikael Simberg, John Biddiscombe and Victor Holanda Rusu (CSCS)
+easyblock = 'CMakeMake'
+
+name = 'HPX'
+version = '1.3.0'
+checksums = ['cd34da674064c4cc4a331402edbd65c5a1f8058fb46003314ca18fa08423c5ad']
```
Comment (lucamar) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-nonetworking.eb]:
```suggestion
versionsuffix = '-nonetworking'
checksums = ['cd34da674064c4cc4a331402edbd65c5a1f8058fb46003314ca18fa08423c5ad']
```

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10.eb [hunk 3]
```diff
@@ -32,17 +32,10 @@ sources = ['%(version)s.tar.gz']
...
 
-apexopts = "-DHPX_WITH_PAPI=ON -DHPX_WITH_APEX=ON -DHPX_WITH_APEX_TAG=v2.1.3 -DAPEX_WITH_PAPI=ON -DAPEX_WITH_OTF2=ON -DOTF2_ROOT=$EBROOTOTF2 -DPAPI_ROOT=$(dirname $PAT_BUILD_PAPI_LIBDIR) -DCMAKE_INSTALL_PREFIX=%(installdir)s/APEX"
-
-configopts = [
-     commonopts,
-     commonopts + apexopts
-]
-
-modextravars = {'HPX_APEX_PATH' : "%(installdir)s/APEX"}
+configopts = commonopts
```
Comment (lucamar) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10.eb]:
```suggestion
configopts = "-DHWLOC_ROOT=$EBROOTHWLOC -DHPX_WITH_MALLOC=JEMALLOC -DJEMALLOC_ROOT=$EBROOTJEMALLOC -DHPX_WITH_MAX_CPU_COUNT=128 -DHPX_WITH_PARCELPORT_MPI=ON -DHPX_WITH_PARCELPORT_MPI_MULTITHREADED=ON"
```

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10.eb [hunk 4]
```diff
@@ -32,17 +32,10 @@ sources = ['%(version)s.tar.gz']
 
 separate_build_dir = True
 
-build_type = 'RelWithDebInfo'
+build_type = 'Release'
 
 commonopts = "-DHWLOC_ROOT=$EBROOTHWLOC -DHPX_WITH_MALLOC=JEMALLOC -DJEMALLOC_ROOT=$EBROOTJEMALLOC -DHPX_WITH_MAX_CPU_COUNT=128 -DHPX_WITH_PARCELPORT_MPI=ON -DHPX_WITH_PARCELPORT_MPI_MULTITHREADED=ON"
```
Comment (lucamar) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10.eb]:
```suggestion
```

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-nonetworking.eb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+source_urls = ['https://github.com/STEllAR-GROUP/hpx/archive']
+sources = ['%(version)s.tar.gz']
+
+separate_build_dir = True
+
+build_type = 'Release'
+
+commonopts = "-DHWLOC_ROOT=$EBROOTHWLOC -DHPX_WITH_MALLOC=JEMALLOC -DJEMALLOC_ROOT=$EBROOTJEMALLOC -DHPX_WITH_MAX_CPU_COUNT=128 -DHPX_WITH_NETWORKING=OFF"
+
+configopts = commonopts
```
Comment (lucamar) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-nonetworking.eb]:
```suggestion
configopts = "-DHWLOC_ROOT=$EBROOTHWLOC -DHPX_WITH_MALLOC=JEMALLOC -DJEMALLOC_ROOT=$EBROOTJEMALLOC -DHPX_WITH_MAX_CPU_COUNT=128 -DHPX_WITH_NETWORKING=OFF"
```

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-nonetworking.eb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+]
+
+source_urls = ['https://github.com/STEllAR-GROUP/hpx/archive']
+sources = ['%(version)s.tar.gz']
+
+separate_build_dir = True
+
+build_type = 'Release'
+
+commonopts = "-DHWLOC_ROOT=$EBROOTHWLOC -DHPX_WITH_MALLOC=JEMALLOC -DJEMALLOC_ROOT=$EBROOTJEMALLOC -DHPX_WITH_MAX_CPU_COUNT=128 -DHPX_WITH_NETWORKING=OFF"
```
Comment (lucamar) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.10-nonetworking.eb]:
```suggestion
```

**Comment by msimberg** [comment]
This is ready to be merged from my side.

**Pull request merged**